### PR TITLE
Updating examples to include required fields

### PR DESCRIPTION
**Pull request by jrk07**
##### Issue Type:
- Docs Pull Request
##### Plugin Name:

ios_template
##### Summary:

Updated examples to include required fields (host, username).  Also updated src option to be listed as required.

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @jrk07. To the current maintainers, @privateip please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Ping @privateip for review of this proposed docs change.

**Comment by privateip** [comment]
:+1:

**Comment by gregdek** [comment]
Thanks @jrk07 for the PR, and thanks @privateip for the review. Marking for inclusion.

**Comment by resmo** [comment]
if `src` should be required, shouldn't this also be implemented in code here:
https://github.com/ansible/ansible-modules-core/pull/3242/files#diff-61e87b88f94efc3125dbcf29b4a5bd09R166 

``` python
src=dict(required=True, type='path'),
```

?

File: network/ios/ios_template.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
         runtime.  By default the task will first search for the source
         file in role or playbook root folder in templates unless a full
         path to the file is given.
-    required: false
+    required: true
     default: null
```
Comment (bcoca) [inline network/ios/ios_template.py:41]: if required is true, default is irrelevant

**Pull request merged**